apiserver/src/main/java/org/dependencytrack/resources/v1/BomResource.java

```diff
@@ -641,4 +751,81 @@ private static boolean shouldValidate(final Project project) {
                    || (validationMode == BomValidationMode.DISABLED_FOR_TAGS && !doTagsMatch);
         }
     }
+
+    private void createNewProject(String projectName, String projectVersion,
+            List<org.dependencytrack.model.Tag> projectTags, Project parent,
+            boolean isLatestProjectVersion) {
+        try (QueryManager qm = new QueryManager()) {
+            final String trimmedProjectName = StringUtils.trimToNull(projectName);
+            final String trimmedProjectVersion = StringUtils.trimToNull(projectVersion);
+
+            if (isLatestProjectVersion) {
+                final Project oldLatest = qm.getLatestProjectVersion(trimmedProjectName);
+                if (oldLatest != null) {
+                    requireAccess(qm, oldLatest, "Access to the previous latest project version is forbidden");
+                }
+            }
+            Project project = qm.createProject(trimmedProjectName, null,
+                    trimmedProjectVersion, projectTags, parent,
+                    null, null, isLatestProjectVersion, true);
+            Principal principal = getPrincipal();
+            qm.updateNewProjectACL(project, principal);
+        }
+    }
+
+    private static JSONObject getJwks(String jwksUrl) throws IOException, InterruptedException {
+        HttpClient client = HttpClient.newHttpClient();
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(URI.create(jwksUrl))
+                .header("Accept", "application/json")
+                .GET()
+                .build();
+
+        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+
+        if (response.statusCode() != 200) 
+            throw new IOException("Failed to fetch JWKS from URL: " + jwksUrl + ". Status code: " + response.statusCode());
```

Comment:
nit: use string formatting if more than one `+` used for concatenation

```suggestion
            throw new IOException("Failed to fetch JWKS from URL: %s. Status code: %d".formatted(jwksUrl, response.statusCode()));
```